apps/meteor/ee/app/license/server/airGappedRestrictions.ts

```diff
@@ -0,0 +1,44 @@
+import { cronJobs } from '@rocket.chat/cron';
+import { AirGappedRestriction } from '@rocket.chat/license';
+import { Settings } from '@rocket.chat/models';
+import { Meteor } from 'meteor/meteor';
+
+import { notifyOnSettingChangedById } from '../../../../app/lib/server/lib/notifyListener';
+import { settings } from '../../../../app/settings/server';
+import { i18n } from '../../../../server/lib/i18n';
+import { sendMessagesToAdmins } from '../../../../server/lib/sendMessagesToAdmins';
+import { checkAirGappedRestrictions } from './airGappedRestrictionsCheck';
+
+const updateRestrictionSetting = async (remainingDays: number) => {
+	const value = settings.get('Cloud_Workspace_AirGapped_Restrictions_Remaining_Days');
+	if (value === remainingDays) {
+		return;
+	}
+	await Settings.updateValueById('Cloud_Workspace_AirGapped_Restrictions_Remaining_Days', remainingDays);
+	void notifyOnSettingChangedById('Cloud_Workspace_AirGapped_Restrictions_Remaining_Days');
+};
+
+const sendRocketCatWarningToAdmins = async (remainingDays: number) => {
+	const lastDayOrNoRestrictionsAtAll = remainingDays <= 0;
+	if (lastDayOrNoRestrictionsAtAll) {
+		return;
+	}
+	if (remainingDays <= AirGappedRestriction.WARNING_PERIOD_IN_DAYS) {
+		await sendMessagesToAdmins({
+			msgs: async ({ adminUser }) => ({
+				msg: i18n.t('AirGapped_Restriction_Warning', { lng: adminUser.language || 'en', remainingDays }),
+			}),
+		});
+	}
+};
+
+AirGappedRestriction.on('remainingDays', async ({ days }: { days: number }) => {
+	await updateRestrictionSetting(days);
+	await sendRocketCatWarningToAdmins(days);
+});
+
+Meteor.startup(async () => {
+	await cronJobs.add('AirGapped Restrictions Cron', '0 */12 * * *', async () => {
```

Comment:
instead of creating a cronjob for this purpose, I'd suggest we use the same one for statistics (`apps/meteor/server/cron/statistics.ts`) and already use the stats token in case it runs successfully.

Reply:
Only concern I have is if this should be outside the EE folder...

Reply:
oh, too late to answer, I saw @gabriellsh already moved the code.. but my reasoning was that since the default state is "restricted", it is fine using the same cronjob on CE code, because it is required to run otherwise it will be locked up.

Reply:
I already moved it @MarcosSpessatto. We can discuss this in DM if you want

Reply:
Yes, there's also that concern. If the Statistics cron runs after the Restriction cron, we'd be always doing the checks with outdated data, possibly locking a workspace that shouldn't be locked @MarcosSpessatto